File: test/ts/factorial.ts
var x: bigint = 5n;
var f: bigint = fac(x);
console.log(f);

function fac(x: bigint): bigint {
    if (x <= 1n) {
        return 1n;
    }
    var result = x * fac(x - 1n);
    return result;
}

File: test/rs/factorial.rs
fn main() {
    let x:i64 = 5;
    let f:i64 = fac(x);
    println!("{:?}", f);
}

fn fac(x:i64) -> i64 {
    if x <= 1 {
        return 1;
    }
    let result: i64 = x * fac(x -1);
    return result;
}
